### ISSUE TITLE
Dart SDK version does not satisfy the requirement

### ISSUE THREAD

**Issue by Skycocoo** [sass/homebrew-sass#5]
Hi,

When I use (upgraded) Homebrew to install sass with ```brew install --devel sass/sass/sass```:

```bash

==> /usr/local/opt/dart/bin/pub get
Last 15 lines from /Users/[]/Library/Logs/Homebrew/sass/01.pub:
2018-05-29 19:16:17 -0400

/usr/local/opt/dart/bin/pub
get

Resolving dependencies...
Package sass requires SDK version >=1.25.0-dev.7.0 <2.0.0 but the current SDK is 1.24.3.
```

The Dart version is the latest stable release: 1.24.3 as indicated in [here](https://webdev-dartlang-org-dev.firebaseapp.com/tools/sdk#about-release-channels-and-version-strings). Also, as indicated in the [changelog by Dart](https://github.com/dart-lang/sdk/blob/master/CHANGELOG.md): it seems that no releases is between 1.25.0-dev.7.0 and 2.0.0.

Am I misunderstanding something?

----

Perhaps the easies way to work around this is just to [install the released version of sass](https://github.com/sass/dart-sass#standalone) (which is also pretty simple)

**Comment by Skycocoo** [comment]
Perhaps the easies way to walk around this is just to install the released version of sass (which is also pretty simple)

**Comment by nex3** [comment]
Looks like [this didn't fix it](https://twitter.com/MikeRogers0/status/1001871359067279361). I'm not sure how to have this formula depend on the devel version of the Dart formula.

**Comment by apjanke** [comment]
> I'm not sure how to have this formula depend on the devel version of the Dart formula.

There's no way to do this directly in a Homebrew formula.

Instead, you can define your own `sass/sass/dart` or `sass/sass/dart@1.25` formula (with a `conflicts_with "homebrew/core/dart"`) in your tap that installs version 1.25 by default, and have `sass/sass/sass` depend on that instead. Then you can have your own fine-grained control over which Dart development version you're installing, and users that `brew install sass/sass/sass` will either get the right Dart version automatically, or a useful error message.

**Comment by ilovezfs** [comment]
Alternatively, convince the dart developers to promote dart 1.25 to stable.

**Comment by apjanke** [comment]
Here's an example of how you could do a versioned formula: #6.

From browsing through the Dart repo, it doesn't look like Dart 1.25.x is a thing that actually exists? So I don't think it'll get promoted. I'm guessing the current `sass` actually works with Dart 2.x, since that's the current `--devel` version in the formula they were referring to.

**Comment by agam** [comment]
_Note:_  I [ran into this](https://github.com/Homebrew/homebrew-core/issues/28470#issuecomment-393365279), adding and removing `dart` works fine (a possible workaround you can mention for users until this is fixed).

**Comment by rootbeersoup** [comment]
This has been fixed. Sass 1.5.1 now builds fine with Dart 1.24.3. No need for the `--devel` version of Dart.

**Comment by Skycocoo** [comment]
Fantastic! The current installation works for me. I guess this issue should be closed.

**Comment by nex3** [comment]
Making Sass Dart 1-compatible again is just a stopgap measure. I want to keep this open until we're actually releasing against dart dev versions.

**Comment by nex3** [comment]
This was fixed by a5783369ee33d84a9e1ec581fbd2fa7f98dd13bf.
